packages/react-components/react-drawer/src/components/OverlayDrawer/OverlayDrawerSurface/OverlayDrawerSurface.tsx

```diff
@@ -14,7 +14,17 @@ import type { OverlayDrawerSurfaceProps } from './OverlayDrawerSurface.types';
  * OverlayDrawerSurface is a proxy for DialogSurface as is only meant to be used internally for Drawer.
  */
 export const OverlayDrawerSurface: ForwardRefComponent<OverlayDrawerSurfaceProps> = React.forwardRef((props, ref) => {
-  const dialogSurfaceState = useDialogSurface_unstable(props, ref);
+  const dialogSurfaceState = useDialogSurface_unstable(
+    {
+      ...props,
+      /**
+       * Drawer accepts a `div` or `aside` element type, but Dialog only accepts a `div` element type.
+       * We need to cast the ref to a `div` element type to not break Dialog's ref type.
+       */
+      as: props.as as 'div',
```

Comment:
Great comment! might be nice to investigate if we can support `aside` on the Dialog side later on and then just remove this casting.

nit: can you also add a FIXME tag here just so that we're aware that the ideal would be to have this being supported by the Dialog instead of having to cast?!